File: 5.js
var read = require('fs').readFile.bind(null, 'data', {encoding: 'utf8'});
var write = require('fs').writeFileSync.bind(null, 'out');

read(function (e, d) {
	d = d.replace(/\r/g, '');

	var n = parseInt(d.split(' ')[0]);
	var m = parseInt(d.split(' ')[1]);

	var slots = Array(m + 1).join('0').split('').map(function (el) { return parseInt(el);});
	slots[0] = 1;

	for (var i = 1; i < n; i++) {
		var sum = 0;
		for (var j = m - 1; j >= 1; j--) {
			sum += slots[j];
			slots[j] = slots[j - 1];
		}
		slots[0] = sum;
		console.log(slots);
	}

	write(slots.reduce(function (a, b) { 
		console.log(a, '+', b, 'is', a + b);
		return a + b; }));
}); 	

File: 5.py
with open('data') as f:
	params = f.read().split()

	n = int(params[0])
	m = int(params[1])

	print('n', n)
	print('m', m)
	slots = [0 for i in range(0, m)]

	slots[0] = 1

	for i in range(1, n):
		summa = 0
		for j in range(m - 1, 0, -1):
			summa += slots[j]
			slots[j] = slots[j - 1]
		slots[0] = summa
		print(slots)
		print('sum', sum(slots))
		
	with open('out', 'w') as f:
		f.write(str(sum(slots)))


